ravenframework/utils/randomUtils.py

```diff
@@ -85,11 +87,101 @@ def testSampling(self, n=1e5,engine=None):
     stdev = np.std(samples)
     return mean,stdev
 
+
+class CrowRNG:
+  """ Wraps crow RandomClass to make it serializable """
+  def __init__(self, engine=None, seed=None):
+    """
+      Constructor
+      @ In, engine, RandomClass, optional, will wrap the given engine if provided, otherwise a new engine is created
+      @ In, reseed_copies, bool, whether or not recovered copies of the class should use a new seed or not
+      @ Out, None
+    """
+    if engine is None:
+      self._engine = findCrowModule('randomENG').RandomClass()
+    elif isinstance(engine, findCrowModule('randomENG').RandomClass):
+      self._engine = engine
+    else:
+      raise TypeError(f'Object of unknown type {type(engine)} cannot be wrapped by CrowRNG class!')
+
+    if seed is not None:
+      self._seed = abs(int(seed))
+      self._engine.seed(self._seed)
+    else:
+      self._seed = self._engine.get_rng_seed()
+
+  def __getstate__(self):
+    """
+      Get state for serialization
+      @ In, None
+      @ Out, d, dict, object instance state
+    """
+    d = copy.copy(self.__dict__)
+    eng = d.pop('_engine')  # remove RNG engine from class instance
+    d['count'] = eng.get_rng_state()
+    return d
+
+  def __setstate__(self, d):
+    """
+      Set object instance state
+      @ In, d, dict, object state
+      @ Out, None
+    """
+    count = d.pop('count')
+    self.__dict__.update(d)
+    self._engine = findCrowModule('randomENG').RandomClass()  # reinstantiate RNG engine
```

Comment:
It seems to me, you can make engine general enough to handle other cases, like numpy random engine. @j-bryan

Reply:
The question seems to be about the scope of this RNG wrapper class. Originally, I had understood the issue as wanting to implement a thin wrapper of RandomClass in Crow so that could be pickled. The numpy random engine wasn't much of a concern since it those can be pickled already and the randomUtils functions already handle the differences between the numpy and crow engines. However, I could see it being beneficial to make a unified RNG class which would handle both the numpy and Crow RNGs in a uniform way.